collector/wifi_linux.go

```diff
@@ -132,13 +130,27 @@ func NewWifiCollector() (Collector, error) {
 			labels,
 			nil,
 		),
-
-		stat: stat,
 	}, nil
 }
 
 func (c *wifiCollector) Update(ch chan<- prometheus.Metric) error {
-	ifis, err := c.stat.Interfaces()
+	// Package wifi makes no guarantees about concurrency, and it's possible that
+	// two concurrent scrapes could cause it to fail
+	c.mu.Lock()
```

Comment:
This is a little unfortunate, but since I haven't done heavy testing of using multiple netlink sockets with one process, I think this is the safest approach to ensure this collector functions as intended.